src/adapter/src/coord.rs

```diff
@@ -2081,13 +2098,79 @@ pub async fn serve(
     let (bootstrap_tx, bootstrap_rx) = oneshot::channel();
     let handle = TokioHandle::current();
 
-    let initial_timestamps = catalog.get_all_persisted_timestamps().await?;
     let metrics = Metrics::register_into(&metrics_registry);
     let metrics_clone = metrics.clone();
+    let timestamp_oracle_metrics =
+        Arc::new(timestamp_oracle::metrics::Metrics::new(&metrics_registry));
     let segment_client_clone = segment_client.clone();
     let span = tracing::Span::current();
     let coord_now = now.clone();
     let advance_timelines_interval = tokio::time::interval(catalog.config().timestamp_interval);
+
+    // We get the timestamp oracle impl once on startup, to ensure that it
+    // doesn't change in between when the system var changes: all oracles must
+    // use the same impl!
+    let timestamp_oracle_impl = catalog.system_config().timestamp_oracle_impl();
```

Comment:
ooc, did you check that this works like you'd expect (we save the change when LD updates and pick it up on the next restart)? Seems easy enough to verify in a staging env with a build of this PR

Reply:
ping :)

Reply:
I didn't check it with LD, no. I imagine it'll be easy enough to do, though, but I'll first try and get through the rest of the issues :see_no_evil:

Reply:
okay no biggie, worst case we can fix it up in post if it doesn't work the way we think